cloudsync/event.py

```diff
@@ -292,27 +299,56 @@ def _fill_event_path(self, event: Event):
                 event.otype = info.otype
                 event.hash = info.hash
     
-    def _filter_event(self, event: Event) -> bool:
-        # event filtering based on root path
-        if not self._root_path:
+    def _filter_event(self, event: Event, from_walk: bool = False) -> bool:
+        # event filtering based on root path and event path
+        # return True = ignore event (filter it out), False = process event
+        if not self._root_path or not self.provider.events_have_path:
             return False
-        # path not a subpath of root
-        if event.exists and event.path and not self.provider.is_subpath(self._root_path, event.path):
-            lookup_oid = event.prior_oid or event.oid
-            state = self.state.lookup_oid(self.side, lookup_oid)
-            if state:
-                # found in state, so this was moved out of our root -- delete
-                log.debug("change event to delete - moved from relevance to irrelevance: %s", state[self.side].path)
-                event.exists = False
-                return False
-            # not in the state db, so this is a create or a move outside our root -- irrelelvant
-            log.debug("ignore irrelevant move/create: %s", event.path)
-            return True
-        # delete of an item that was not in our state db -- irrelevant
-        elif not event.exists and not self.state.lookup_oid(self.side, event.oid):
-            log.debug("ignore irrelevant delete: %s", event.oid)
-            return True
-        return False
+
+        curr_subpath = self.provider.is_subpath_of_root(event.path)
```

Comment:
don't you need to pass the root to this function?

Reply:
added `provider.is_subpath_of_root()` as a convenience function - and also to [try to] make the provider the single source of truth as to what the root is (though I have not gone all the way in implementing this yet)